cmd/tsconnect/wasm/wasm_js.go

```diff
@@ -323,6 +343,120 @@ func (i *jsIPN) logout() {
 	go i.lb.Logout()
 }
 
+func (i *jsIPN) tcp(host string, port int, config js.Value) interface{} {
+	portInt := strconv.Itoa(port)
+	addr := net.JoinHostPort(host, portInt)
+
+	return makePromise(func() (any, error) {
+		if config.Type() != js.TypeFunction {
+			return nil, errors.New("invalid config")
+		}
+
+		jsTCPSession, err := i.NewJsTCPSession(addr, config)
+		if err != nil {
+			return nil, err
+		}
+
+		jsObjectWrapper := jsTCPSession.jsWrapper()
+		return jsObjectWrapper, nil
+	})
+
+}
+
+type jsTCPSession struct {
+	jsIPN       *jsIPN
+	conn        net.Conn
+	writeBuffer []byte
+	readBuffer  []byte
+}
+
+func (i *jsIPN) NewJsTCPSession(addr string, config js.Value) (*jsTCPSession, error) {
+	if config.Type() != js.TypeFunction {
+		return nil, errors.New("Invalid callback received")
+	}
+
+	const timeoutSeconds time.Duration = 5
```

Comment:
Ah my bad wrong variable name but I do multiply it later at line 379 will fix this. The code atm tho is not really polished overall as I wanted some feedback first, if this would even be wanted in the package.